lms/services/canvas_api.py

```diff
@@ -65,6 +69,35 @@ def get_refreshed_token(self, refresh_token):
 
         return new_access_token
 
+    def authenticated_users_sections(self, course_id):
+        """
+        Return the authenticated user's sections for the given course_id.
+
+        Send an HTTP request to the Canvas API to get the list of sections and
+        return it.
+
+        :arg course_id: the Canvas course_id of the course to look in
+        :type course_id: str
+
+        :raise lms.services.CanvasAPIAccessTokenError: if we can't get the list
+            of sections because we don't have a working Canvas API access token
+            for the user
+        :raise lms.services.CanvasAPIServerError: if we do have an access token
+            but the Canvas API request fails for any other reason
+
+        :return: a list of raw section dicts as received from the Canvas API
+        :rtype: list(dict)
+        """
```

Comment:
Adding support for a new Canvas API endpoint involves adding code in three separate places:

1. In the Canvas API helper (above)
2. In the Canvas API service (here)
3. In the Canvas API validation schemas (below)

It'd be nice if these three separate locations could be brought together so that adding a new endpoint only involved adding code in one place, and didn't involve having to explain the same thing three times in three separate docstrings.

The code design is perfectly workable though and I'm more focused on making forward progress with the sections feature right now, so I'm not going to refactor the code now.